src/components/GameProvider/index.tsx

```diff
@@ -147,7 +147,11 @@ const GameProvider = (props: { children: ReactNode }) => {
 
   const addPurchasedCard = useCallback(
```

Comment:
feels like this method is a bit dangerous now - it's up to consumers to know check gold counts before calling this method, or gold amount may go negative.

there are two ways we can go: renaming this to `purchaseCard`, and making it do a gold check first. If it fails, it returns false and doesn't add the purchased card. there are other checks we may want to do here too like checking the required level. there may be cases where we want to directly add cards to a player's inventory though. for that case we could also add an optional parameter of `ignoreCost` or something: if provided, it doesnt do a gold cost check or deduction.

alternatively, we dont do any checks or deductions, and create a new purchase method used in the training window that does checks and deductions; and maybe add a comment to this `addPurchasedCard` method to explain it's a direct write to the character state and doesnt do any validation.

what do you think

Reply:
+1 to splitting this method into to one with and without handling the purchase cost. I would lean slightly towards having `purchaseCard` and `takeCardWithoutValidation` methods exposed here because its more explicit.

There's cases where you will need to take a card without actually purchasing it (like quest rewards?), so I think a simple solution is to just provide both "Purchase" and "Take" options in the carousel UI, where the latter just gives the player the card with no validation.